src/app/features/noah-playground/components/map-playground/map-playground.component.ts

```diff
@@ -271,39 +272,46 @@ export class MapPlaygroundComponent implements OnInit, OnDestroy {
         coordinates[0] += e.lngLat.lng > coordinates[0] ? 360 : -360;
       }
 
-      popUp
-        .setLngLat(coordinates)
-        .setHTML(
+      const sensors = {
+        ...this.store.state.sensors,
+      };
+
+      if (sensors.shown) {
```

Comment:
Let's add a getter in the playground service instead of accessing the store directly from the component. We just want to avoid the components getting full context of what the state is and apply changes there accidentally. Having a streamlined flow of where the data comes and where it should go may have additional overhead for us but that would help us avoid having spaghetti code in the future :)

Reply:
Ok. Noted on this :)